### DISCUSSION TITLE
Can't endpoint use `export default` syntax?

### DISCUSSION THREAD

**Discussion by cdwmhcc** [directus/directus#17042] [Archive]
extensions/endpoint/test/index.`ts`

``` ts
export default (router) => {
	router.get('/', (req, res) => res.send('Hello, World!'));
};
```
It does not work.

extensions/endpoint/test/index.`js`
``` js
module.exports = (router) => {
  router.get("/", async (req, res) =>res.send('Hello, World!'));
};
```
it worked

**Comment by br41nslug** [comment]
You can but for TypeScript or ESM you'll have to use the extension SDK https://docs.directus.io/extensions/creating-extensions.html#building-your-extension

**Reply by cdwmhcc** [reply]
Thanks for your reply. @rijkvanzanten Is it possible to add automatic conversion? What do you think?

**Reply by rijkvanzanten** [reply]
It is not possible to add automatic conversion, as they're inherently different things. `export default` is ESM whereas `module.exports` is CJS. The two aren't necessarily compatible in Node.
